src/k8s/pkg/k8sd/app/hooks_remove.go

```diff
@@ -124,19 +125,21 @@ func (a *App) onPreRemove(ctx context.Context, s state.State, force bool) (rerr
 		log.Error(err, "Failed to unmark node as worker")
 	}
 
-	log.Info("Stopping worker services")
-	if err := snaputil.StopWorkerServices(ctx, snap); err != nil {
-		log.Error(err, "Failed to stop worker services")
-	}
-
 	log.Info("Cleaning up control plane certificates")
 	if _, err := setup.EnsureControlPlanePKI(snap, &pki.ControlPlanePKI{}); err != nil {
 		log.Error(err, "failed to cleanup control plane certificates")
 	}
 
-	log.Info("Stopping control plane services")
-	if err := snaputil.StopControlPlaneServices(ctx, snap); err != nil {
-		log.Error(err, "Failed to stop control-plane services")
+	if _, ok := cfg.Annotations[apiv1.AnnotationSkipStopServicesOnRemove]; !ok {
```

Comment:
you mentioned that a stopped `kubelet` was the root cause of this, so I imagine we should be able to notice this node deletion problem on removing worker nodes (in CAPI) as well, right?

Reply:
As discussed offline: Added worker node propagation for annotations, so that this also works on a worker. 
Added an extra test step on the integration test to verify this.